store/mockstore/mocktikv/rpc.go

```diff
@@ -971,11 +971,11 @@ func (c *RPCClient) SendRequest(ctx context.Context, addr string, req *tikvrpc.R
 	case tikvrpc.CmdUnsafeDestroyRange:
 		panic("unimplemented")
 	case tikvrpc.CmdRegisterLockObserver:
-		panic("unimplemented")
+		return nil, errors.New("unimplemented")
 	case tikvrpc.CmdCheckLockObserver:
-		panic("unimplemented")
+		return nil, errors.New("unimplemented")
 	case tikvrpc.CmdRemoveLockObserver:
-		panic("unimplemented")
+		return nil, errors.New("unimplemented")
 	case tikvrpc.CmdPhysicalScanLock:
 		panic("unimplemented")
```

Comment:
Should this return an error instead of panic?

Reply:
It may panic during tests. Returning error can make it fall back to legacy mode to finish the test. I need to add more unit tests to specifically cover the new logic later.

Reply:
I mean `PhysicalScanLock` is also an API depended by green GC and can be also used to make it fallback to the old logic...
(Although it should be `registerLockObservers` that makes it fallback in fact)

Reply:
Oh yes. I'll change it.